mixer/pkg/runtime/config/yaml/resolver.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2018 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package yaml
+
+import (
+	"github.com/gogo/protobuf/protoc-gen-gogo/descriptor"
+
+	"istio.io/istio/mixer/tools/codegen/pkg/modelgen"
```

Comment:
I don't think a runtime component should take a dependency on a tool package. Can we refactor this? If the code in the tool package is truly generic, then let's move it to the main pkg directory.

Reply:
+1